middleware.tsx

```diff
@@ -0,0 +1,33 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { Cookies, SEVEN_DAYS } from '@/constants';
+
+export const config = {
+  matcher: '/',
+};
+
+export async function middleware(req: NextRequest) {
+  const res = NextResponse.next();
+
+  const getCountry = req.cookies.get('country')?.value ?? '';
+
+  const userIp = req.ip ?? req.headers.get('x-real-ip');
+
+  if (userIp) {
+    res.cookies.set('user-ip', userIp, {
+      httpOnly: false,
+    });
+  }
+
+  if (!getCountry) {
+    try {
+      const response = await fetch(`http://ip-api.com/json/${userIp}`);
```

Comment:
The ips are all empty. I read about it, and I guess there is no way to get from simple localhost. Nonetheless, we still can train this task. if you just call that `http://ip-api.com/json` endpoint, it will detect IP of the machine and return the country for it. I think it will be enough. The main of the ticket is to train cookies, not IP detection)

![image](https://github.com/Alex-2kZharkov/oldboy-barbershop/assets/66321831/06973c03-f66b-434e-b84c-2d5840f94024)